src/lite.h

```diff
@@ -139,10 +140,15 @@ int     whichp     (const char *cmd);
  */
 static inline int touch(const char *path)
 {
-	if (utimensat(AT_FDCWD, path, NULL, 0)) {
-		if (errno == ENOENT)
-			return mknod((path), S_IFREG|0644, 0);
-		return -1;
+	int fd;
+
+	if (utime(path, NULL)) {
+		if (errno != ENOENT)
+			return -1;
+		fd = creat(path, S_IRUSR|S_IWUSR|S_IRGRP|S_IROTH);
+		if (fd < 0)
+			return -1;
+		close(fd);
```

Comment:
This I cannot approve, unfortunately.  We cannot rely on the poor granularity of `utime()` on the platforms that use `touch()` in Finit. Sorry, but we've had some really nasty bugs with this in the past.  Please replace this with [inadyn/lib/utimensat.c ](https://github.com/troglobit/inadyn/blob/master/lib/utimensat.c), which I should've added myself years ago to libite, but never got around to.

Not really sure what's the most appropriate way, but since it's in POSIX.1-2008 I think as a library replacement is the best way.  I.e., exactly like it's done in In-a-dyn (and mdnsd, ssdp-responder, bridged, mini-snmpd, mrouted, smcroute) and other's I've run into this problem with in the past.